app/models/freightage_calc.rb

```diff
@@ -0,0 +1,72 @@
+class FreightageCalc
+  def calculate_freightage(shipping_co, distance, weight, volume)
+    value = find_price_setting(shipping_co, weight, volume).distance_value * distance
+    days = find_shipment_time_setting(shipping_co, distance).shipping_days
+    { price: value, shipment_days: days }
+  end
+
+  def find_by_distance(distance)
+    valid_settings = @shipment_time_settings.select do |sts|
+      sts.within_distance_range?(distance)
+    end
+    return [find_closest(distance)] if valid_settings.empty?
+
+    valid_settings
+  end
+
+  def find_by_volume(product_volume)
+    valid_price_settings = @price_settings.select do |price_setting|
+      price_setting.within_volume_range?(product_volume)
+    end
+
+    return valid_price_settings unless valid_price_settings.empty?
+
+    closest_min = @price_settings.min_by { |ps| (ps.volume_min - product_volume).abs }
+    closest_max = @price_settings.min_by { |ps| (ps.volume_max - product_volume).abs }
+
+    return [closest_min] if closest_min == closest_max
+
+    [closest_min, closest_max]
+  end
+
+  def find_by_weight(product_weight)
+    valid_price_settings = @price_settings.select do |price_setting|
+      price_setting.within_weight_range?(product_weight)
+    end
+    return valid_price_settings unless valid_price_settings.empty?
+
+    closest_min = @price_settings.min_by { |ps| (ps.weight_min - product_weight).abs }
+    closest_max = @price_settings.min_by { |ps| (ps.weight_max - product_weight).abs }
+
+    return [closest_min] if closest_min == closest_max
+
+    [closest_min, closest_max]
+  end
+
+  def find_closest(distance)
+    @shipment_time_settings.min_by do |setting|
+      difference_min = (setting.distance_min - distance).abs
+      difference_max = (setting.distance_max - distance).abs
+
+      difference_max < difference_min ? difference_max : difference_min
+    end
+  end
+
+  def find_price_setting(shipping_co, product_weight, product_volume)
+    @price_settings = shipping_co.price_settings
+    price_settings_with_weight = find_by_weight(product_weight)
+    price_settings_with_volume = find_by_volume(product_volume)
+
+    best_price_setting = (price_settings_with_weight & price_settings_with_volume).min_by(&:distance_value)
+
+    return best_price_setting unless best_price_setting.nil?
```

Comment:
```suggestion
    return best_price_setting if best_price_setting.present?
```

essa aqui é uma mania então fiquem a vontade pra fazer como quiser, eu costumo usar pouco unless pra dar mais legibilidade usando ifs com métodos que dizem mais. Então é uma pequena sugestão muito pessoal que vocês podem ou não gostar

Reply:
Eu gostei! Fica mais humano ler `present?` ao invés de `nil?` também :)

Alterado, obrigada pela sugestão